addons/website_slides/controllers/main.py

```diff
@@ -1331,9 +1464,19 @@ def _get_channels(self, **kwargs):
             channels = tools.lazy(lambda: request.env['slide.channel'].browse(int(kwargs['channel_id'])))
         return channels
 
+    def _get_invite_error_msg(self, **kwargs):
+        invite_error_msgs = {
+            'expired': _('This invitation link has expired.'),
+            'hash_fail': _('This invitation link has an invalid hash.'),
+            'no_channel': _('This course does not exist.'),
+            'no_rights': _('You do not have the rights to access this course.'),
+            'partner_fail': _('This invitation link is not for this partner.'),
+        }
+        return invite_error_msgs.get(kwargs.get('invite_state', 'wrong_state'), '')
```

Comment:
This one may be personal, but it's one less step to follow.

```suggestion
            'partner_fail': _('This invitation link is not for this partner.'),
            'unknown_state`: '',
        }[kwargs.get('invite_state', 'unknown_state')]
```